batch.go

```diff
@@ -144,7 +144,10 @@ func (batch *Batch) Read(b []byte) (int, error) {
 // it is less memory-efficient than Read, but has the advantage of never
 // failing with io.ErrShortBuffer.
 func (batch *Batch) ReadMessage() (Message, error) {
-	msg := Message{}
+	msg := Message{
+		Topic:     batch.conn.topic,
+		Partition: batch.conn.partition,
+	}
```

Comment:
Can we set this only on success? Otherwise when an error is returned it’s not the zero-value if the message which is being returned anymore.